### PR TITLE
Cleanup the rest of codegen_llvm

### PR DESCRIPTION
**Pull request by ljedrz**
- improve common patterns
- convert string literals with `to_owned`
- remove explicit `return`s
- whitespace & formatting improvements

### CODE REVIEW

**Comment by ljedrz** [comment]
It seems @michaelwoerister might currently be unavailable; r? @varkor

File: src/librustc_codegen_llvm/debuginfo/metadata.rs [hunk 1]
```diff
@@ -163,10 +163,10 @@ impl TypeMap<'ll, 'tcx> {
     fn get_unique_type_id_of_type<'a>(&mut self, cx: &CodegenCx<'a, 'tcx>,
                                       type_: Ty<'tcx>) -> UniqueTypeId {
         // Let's see if we already have something in the cache
-        match self.type_to_unique_id.get(&type_).cloned() {
-            Some(unique_type_id) => return unique_type_id,
-            None => { /* generate one */}
-        };
+        if let Some(unique_type_id) = self.type_to_unique_id.get(&type_).cloned() {
+            return unique_type_id
```
Comment (varkor) [inline src/librustc_codegen_llvm/debuginfo/metadata.rs]: `return` as part of a statement should [end with a semicolon](https://github.com/nrc/rfcs/blob/style-guide/style-guide/statements.md#expressions-in-statement-position). (Here and in the other changes.)

File: src/librustc_codegen_llvm/debuginfo/source_loc.rs [hunk 2]
```diff
@@ -56,11 +56,8 @@ pub fn set_source_location(
 /// switches source location emitting on and must therefore be called before the
 /// first real statement/expression of the function is codegened.
 pub fn start_emitting_source_locations(dbg_context: &FunctionDebugContext<'ll>) {
-    match *dbg_context {
-        FunctionDebugContext::RegularContext(ref data) => {
-            data.source_locations_enabled.set(true)
-        },
-        _ => { /* safe to ignore */ }
+    if let FunctionDebugContext::RegularContext(ref data) = *dbg_context {
+        data.source_locations_enabled.set(true)
```
Comment (varkor) [inline src/librustc_codegen_llvm/debuginfo/source_loc.rs]: Same here.

**Review by varkor — COMMENTED** [review]
Looks good to me apart from the semicolon nits.

File: src/librustc_codegen_llvm/debuginfo/metadata.rs (hunk 1 above)
Comment (ljedrz) [reply src/librustc_codegen_llvm/debuginfo/metadata.rs]: I wasn't aware of that; thanks, I'll update the PR shortly.

**Comment by ljedrz** [comment]
@varkor thanks for the review! Semicolons added.

**Pull request merged**